crates/driver/src/infra/observe/mod.rs

```diff
@@ -131,33 +132,33 @@ pub fn score(solver: &solver::Name, settlement: &Settlement, score: &solution::S
     tracing::info!(
         %solver,
         solutions = ?settlement.solutions(),
-        settlement_id = ?settlement.id,
+        auction_id = ?settlement.auction_id,
         score = score.0.to_f64_lossy(),
         "settlement scored"
     );
 }
 
 /// Observe that the settlement process is about to start.
-pub fn settling(solver: &solver::Name, id: settlement::Id) {
-    tracing::trace!(%solver, ?id, "settling");
+pub fn settling(solver: &solver::Name, auction_id: Option<auction::Id>) {
```

Comment:
I think `auction_id` should probably non-optional in `settling()` and `settled()`.
If we try to settle a solution without an `auction_id` probably some invariant has been violated.

Reply:
Yes, in case this is `None` the endpoint will return an error. But I think that this should still be logged.

Reply:
But it definitely shouldn't be optional in `observe::settled`.